adal/src/main/java/com/microsoft/aad/adal/TokenCacheAccessor.java

```diff
@@ -353,22 +358,28 @@ void updateTokenCacheUsingCommonCache(final AuthenticationRequest request, final
             config.setAuthorityUrl(new URL(mAuthority));
         }
 
-        AzureActiveDirectoryOAuth2Strategy strategy = ad.createOAuth2Strategy(config);
-
-        AzureActiveDirectoryAuthorizationRequest.Builder aadAuthRequestBuilder = new AzureActiveDirectoryAuthorizationRequest.Builder();
-        aadAuthRequestBuilder
-                .setClientId(request.getClientId())
-                .setResource(request.getResource())
-                .setScope(request.getResource())
-                .setRedirectUri(request.getRedirectUri())
-                .setLoginHint(request.getLoginHint())
-                .setCorrelationId(request.getCorrelationId());
+        AzureActiveDirectoryOAuth2Strategy strategy = null;
+        try {
+            strategy = ad.createOAuth2Strategy(config,
+                    new AndroidCommonComponents(mContext));
+
+            AzureActiveDirectoryAuthorizationRequest.Builder aadAuthRequestBuilder = new AzureActiveDirectoryAuthorizationRequest.Builder();
+            aadAuthRequestBuilder
+                    .setClientId(request.getClientId())
+                    .setResource(request.getResource())
+                    .setScope(request.getResource())
+                    .setRedirectUri(request.getRedirectUri())
+                    .setLoginHint(request.getLoginHint())
+                    .setCorrelationId(request.getCorrelationId());
+
+            if (null != mAuthority) {
+                aadAuthRequestBuilder.setAuthority(new URL(mAuthority));
+            }
 
-        if (null != mAuthority) {
-            aadAuthRequestBuilder.setAuthority(new URL(mAuthority));
+            mCommonCache.save(strategy, aadAuthRequestBuilder.build(), tokenResponse);
+        } catch (final ClientException e) {
+            throw ADALError.fromCommon(e);
```

Comment:
Is this `fromCommon` thing a new API? or was it already there?

Reply:
not new. already exist :)